include/z64.h

```diff
@@ -1743,14 +1743,17 @@ typedef struct {
     /* 0x100 */ u16 acCodes[256];
 } JpegHuffmanTableOld; // size = 0x300
 
-typedef struct {
+typedef union {
+    struct {
     /* 0x00 */ u32 address;
     /* 0x04 */ u32 mbCount;
     /* 0x08 */ u32 mode;
     /* 0x0C */ u32 qTableYPtr;
     /* 0x10 */ u32 qTableUPtr;
     /* 0x14 */ u32 qTableVPtr;
-    /* 0x18 */ char unk_18[0x8];
+    /* 0x18 */ u32 mbSize; // This field is used by the microcode to save the macroblock size during a yield
+    };
+    long long int force_structure_alignment;
```

Comment:
then the 4 pad bytes at 0x1C come from `force_structure_alignment`, that may be a bit sneaky?

also this jpeg ucode was never released was it, the manual part on it sure isn't entirely accurate

Reply:
I don't see much of an issue with the struct size being padded out by `force_structure_alignment`, I think that's more so part of the point. It has to be a multiple of 8 bytes for the DMA to function properly after all.
The JPEG library described in the manual seems to be related but not the same. I don't really know much about the library described in the manual since afaik we don't have any unlinked version of it to investigate any debugging info.

Reply:
I mean are you sure that word is padding then, since up until now it's been documented as unk_
How did you figure out/learn about this `mbSize` field?

Reply:
I have an almost 100% documented disassembly of the jpeg microcode